### ISSUE TITLE
Auto-instrumentation with JMX

### ISSUE THREAD

**Issue by sarkisjad** [open-telemetry/opentelemetry-java-instrumentation#9336]
### Describe the bug

I am trying to implement auto-instrumentation using the [jmx-metrics](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/jmx-metrics/javaagent)
My aim is to emit a metric on a simple Java application for the thread count. 
When I run the command (shown in 'Steps to reproduce') and create a rules file (as instructed in the documentation), the metric described in said rules file is not emitted even when I leave the application running for a while. Yet, when I open jconsole, it gets emitted successfully. I am assuming it is because I am launching a trigger that will help the process out. 

### Steps to reproduce

Here is the following command launched on my Windows terminal: 
`java -javaagent:./opentelemetry-javaagent.jar
    -Dotel.instrumentation.common.default-enabled=false
    -Dotel.jmx.discovery.delay=1000
    -Dotel.jmx.config=config_file.yml
    -jar my-application.jar
`

With the config_file.yml being:
```
rules:
  - bean: java.lang:type=Threading
    mapping:
      ThreadCount:
        metric: my.own.jvm.thread.count
        type: gauge
        desc: The current number of threads
        unit: "1"
```

### Expected behavior

One of the metrics emitted should be the following: 
`[{"name":"my.own.jvm.thread.count","description":"The current number of threads","unit":"1","sum":{"dataPoints":[{"startTimeUnixNano":"1692280364935098100","timeUnixNano":"1692280484952967900","asInt":"23"}],"aggregationTemporality":2}}]`

It is emitted only when we launch jconsole. 

### Actual behavior

No such metric is emitted when we leave the application to run alone, without any additional help (jconsole).

### Javaagent or library instrumentation version

v1.29.0 (latest as of date)

### Environment

**JDK**: 17.0.2
**OS**: 

### Additional context

I have opentelemetry configured. It is listening to the default port (4317) and successfully capturing metrics and logging it in a random text file, as per my configuration. 
We have no issues on that side.

**Comment by mateuszrzeszutek** [comment]
Ping @PeterF778 -- is this something you could take a look at?

**Comment by PeterF778** [comment]
Yes, please assign it to me.
Thanks

________________________________

**Comment by PeterF778** [comment]
Providing a fix as PR https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9362

**Comment by breedx-splk** [comment]
Looks like the fix landed in #9362 but this wasn't auto closed. Closing now, but please reopen if I have misunderstood.
